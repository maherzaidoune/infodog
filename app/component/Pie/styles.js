import {StyleSheet, Dimensions} from 'react-native';

export const styles = StyleSheet.create({
  infoContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  infoText: {
    fontWeight: '600',
    fontSize: 17,
  },
});
