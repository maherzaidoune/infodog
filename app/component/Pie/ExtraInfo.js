import React from 'react';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { styles } from './styles';

/**
 * Information to be shown inside the Circular progreee
 * @module
 */
export default function ExtraInfo({ fill, icon, color }) {
  return (
    <View style={styles.infoContainer}>
      {icon && <Icon name={icon} size={21} color={color} />}
      <Text style={styles.infoText}>
        {fill}
        %
      </Text>
    </View>
  );
}
