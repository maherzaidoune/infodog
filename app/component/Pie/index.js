import React from 'react';
import {View, Text, Dimensions} from 'react-native';
import {AnimatedCircularProgress} from 'react-native-circular-progress';
import {colors} from '../../constants/colors';
import ExtraInfo from './ExtraInfo';
import {styles} from './styles';

const index = ({fill = 0, icon, color}) => {
  return (
    <AnimatedCircularProgress
      size={120}
      width={15}
      fill={fill}
      rotation={0}
      lineCap={'round'}
      tintColor={color}
      backgroundColor={colors.secondary}
      onAnimationComplete={() => console.log('onAnimationComplete')}>
      {() => <ExtraInfo fill={fill} icon={icon} color={color} />}
    </AnimatedCircularProgress>
  );
};

export default React.memo(index);
