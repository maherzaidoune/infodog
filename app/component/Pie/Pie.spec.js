import React from 'react';
import Pie from '.';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Pie fill={50} icon={'test'} color={'red'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
