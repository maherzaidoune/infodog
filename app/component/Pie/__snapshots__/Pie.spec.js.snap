// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={Object {}}
>
  <RNSVGSvgView
    bbHeight={120}
    bbWidth={120}
    focusable={false}
    height={120}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 120,
          "width": 120,
        },
      ]
    }
    width={120}
  >
    <RNSVGGroup>
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            -0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGPath
          d="M 59.96701327930773 7.500010363084286 A 52.5 52.5 0 1 0 60 7.5"
          fill={0}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={4293322470}
          strokeLinecap={1}
          strokeWidth={15}
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 45,
        "height": 90,
        "justifyContent": "center",
        "left": 15,
        "overflow": "hidden",
        "position": "absolute",
        "top": 15,
        "width": 90,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "red",
              "fontSize": 21,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ?
      </Text>
      <Text
        style={
          Object {
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        50
        %
      </Text>
    </View>
  </View>
</View>
`;
